basic/src/main/java/com/c8y/ms/templates/basic/service/MeasurementService.java

```diff
@@ -0,0 +1,64 @@
+package com.c8y.ms.templates.basic.service;
+
+import com.c8y.ms.templates.basic.utils.RevertQueryParameter;
+import com.cumulocity.model.idtype.GId;
+import com.cumulocity.rest.pagination.RestPageRequest;
+import com.cumulocity.rest.representation.measurement.MeasurementRepresentation;
+import com.cumulocity.sdk.client.SDKException;
+import com.cumulocity.sdk.client.measurement.MeasurementApi;
+import com.cumulocity.sdk.client.measurement.MeasurementFilter;
+import com.google.common.base.Optional;
+import org.apache.commons.lang.StringUtils;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class MeasurementService {
+    private static final Logger LOG = LoggerFactory.getLogger(MeasurementService.class);
+
+    private final MeasurementApi measurementApi;
+
+    public MeasurementService(MeasurementApi measurementApi) {
+        this.measurementApi = measurementApi;
+    }
+
+    /**
+     * Query the latest measurement for a specific device. This requires to set the
+     * date and also to provide the query parameter "revert=true". For the revert
+     * query parameter a custom class RevertQueryParameter has been implemented
+     * which is used on the list. If you want to query for a specific type you need
+     * to add the type as a filter to the MeasurementFilter
+     *
+     * @param deviceId id of device for which the latest measurement should be queried
+     * @return the latest measurement as MeasurementRepresentation
+     */
+    public Optional<MeasurementRepresentation> getLatestMeasurement(final String deviceId) {
+        if (StringUtils.isEmpty(deviceId)) {
+            return Optional.absent();
+        }
+
+        final DateTime dateFrom = new DateTime(2020, 1, 1, 0, 0);
+        final DateTime dateNow = new DateTime();
+        final MeasurementFilter filter = new MeasurementFilter().bySource(new GId(deviceId)).byDate(dateFrom.toDate(),
+                dateNow.toDate());
+
+        try {
+            final List<MeasurementRepresentation> measurements = measurementApi.getMeasurementsByFilter(filter)
+                    .get(RestPageRequest.MAX_PAGE_SIZE, new RevertQueryParameter()).getMeasurements();
```

Comment:
I would not create RevertQueryParameter for each measurement, it could be static final class instance in MeasurementService or a singleton.

Reply:
changed to singleton